treeherder/client/thclient/client.py

```diff
@@ -701,6 +708,40 @@ def _post_json(self, project, endpoint, oauth_key, oauth_secret, jsondata,
                              timeout=timeout)
         resp.raise_for_status()
 
+    def _get_json(self, project, endpoint, timeout=None):
+
+        if timeout is None:
+            timeout = self.timeout
+        response = requests.get(self._get_uri(project, endpoint), timeout=timeout)
+        response.raise_for_status()
+        return response.json()
+
+    def query_resultset_api(self, project, **params):
+        """
+        Gets resultsets from project, filtered by parameters
+
+        By default this method will just return the latest 10 result sets (if they exist)
+
+        :param project: project (repository name) to query data for
+        :param params: keyword arguments to filter results
+        """
+
+        string = '&'.join("%s=%s" % (key, val) for key, val in params.iteritems())
```

Comment:
You don't need to encode the querystring parameters yourself, let requests handle that. See http://docs.python-requests.org/en/latest/user/quickstart/#passing-parameters-in-urls